### ISSUE TITLE
Add MemAvailable per node

### ISSUE THREAD

**Issue by antoine-morvan** [numactl/numactl#210]
Hello, 

I am running into an issue where I need the memory available per node. For meminfo this was added into the kernel a while back (https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773) and it now shows the amount of memory we can allocate before failure: 
```bash
$ cat /proc/meminfo
MemTotal:       601685376 kB
MemFree:        265489792 kB
MemAvailable:   551693760 kB #       << here
Buffers:         2575360 kB
Cached:         314235008 kB
SwapCached:            0 kB
Active:         20044160 kB
# [...]
```

However numastat does not display such information :
```bash
$ numastat -V
numastat version: 20130723: Jan 30 2024
$ numastat -mc

Per-node system memory usage (in MBs):
                 Node 0 Node 1 Node 2 Node 3 Node 4 Node 5 Node 6 Node 7 Node 8
                 ------ ------ ------ ------ ------ ------ ------ ------ ------
MemTotal         490303  97280      0      0      0      0      0      0      0
MemFree          161993  97272      0      0      0      0      0      0      0
MemUsed          328310      8      0      0      0      0      0      0      0
SwapCached            0      0      0      0      0      0      0      0      0
Active            19574      0      0      0      0      0      0      0      0
# [...]
```

Any hope this can added in numastat ?

Best regards,
Antoine.

**Comment by luochenglcs** [comment]
Sorry, currently, the kernel does not provide an interface to directly obtain the each node's available memory, but we can roughly estimate from each node's MemFree, Active(file), Inactive(file), SReclaimable.

You can see each node's meminfo form /sys/bus/node/devices/nodex, if your kernel is configured with CONFIG_NUMA.

**Comment by luochenglcs** [comment]
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773 This submission estimates the system's available memory.

**Comment by andikleen** [comment]
I think to add it I would like it to be in the /sys/device/system/node/node*/meminfo instead of adding heuristics.

Can you perhaps send a kernel patch to add it to the kernel people? It should be simple enough I hope.

**Comment by luochenglcs** [comment]
Yes, I have sent kernel patch, please review it.  Link: https://lore.kernel.org/linux-mm/20240204083414.107799-1-luochunsheng@ustc.edu/
